sympy/physics/continuum_mechanics/beam.py

```diff
@@ -1186,6 +1186,143 @@ def max_deflection(self):
         else:
             return None
 
+
+    def bending_stress(self, y, m):
+        """
+        Returns bending stress of any point(or layer) in the beam cross-section at a distance `y`
+        from the neutral axis.
+
+        Parameters
+        ==========
+        y : Sympifiable
+            A symbol or a value representing the vertical distance of the required point(or layer)
+            on the cross-section from the neutral axis.
+        m : Sympifiable
+            A symbol or a value representing the bending moment at the required cross-section of
+            the beam.
+
+        Example
+        =======
+        There is a beam of length 6 m with a u.d.l of 40 KN/m over its length. The beam has a rectangular
+        cross-section(0.3 x 0.6 m) with moment of inertia. The bending stress at a point
+        0.2 m above the neutral axis and at a cross-section which is 4 meters from the start of the beam
+        has to be determined.
+
+        >>> from sympy.physics.continuum_mechanics.beam import Beam
+        >>> from sympy import symbols, Symbol
+        >>> E, I = symbols('E, I')
+        >>> R1, R2 = symbols('R1, R2')
+        >>> b = Beam(6, E, I)
+        >>> b.apply_load(R1, 0, -1)
+        >>> b.apply_load(40, 0, 0, end=6)
+        >>> b.apply_load(R2, 6, -1)
+        >>> b.solve_for_reaction_loads(R1, R2)
+        >>> moment = b.bending_moment()
+        >>> x = Symbol('x')
+        >>> moment = abs(moment.subs(x, 4))
+        >>> b.bending_stress(0.2, moment)
+        32.0/I
```

Comment:
On the face of it, this API design seems awkward.

The simplest thing would be to output a function of the beam's length (x) and distance from the neutral axis (y). The user can then use xreplace, subs, and/or evalf to get a specific value.

Reply:
For example:

```
>>> beam.bending_moment()
F*(l-x)
>>> beam.bending_stress()
F*(l-x)*y/I
```